lib/cylc/scheduler.py

```diff
@@ -388,9 +388,9 @@ def command_set_stop_cleanly(self, kill_active_tasks=False):
         """Stop job submission and set the flag for clean shutdown."""
         SuiteProcPool.get_inst().stop_job_submission()
         TaskProxy.stop_sim_mode_job_submission = True
-        if kill_active_tasks:
-            self.pool.kill_active_tasks()
         self.shut_down_cleanly = True
+        self.kill_on_shutdown = kill_active_tasks
+        self.next_kill_issue = datetime.datetime.utcnow()
```

Comment:
I prefer `time.time` to `datetime.datetime.utcnow`. Although their run times are both insignificantly fast, the former is still about 10 times faster. It also makes it easier for the logic later on when you are adding 10 seconds, - you can just do `time.time() + 10.0` instead of have to wrap the delta with `datetime.timedelta`.